packages/tpetra/core/src/Tpetra_DirectoryImpl_def.hpp

```diff
@@ -431,10 +431,25 @@ namespace Tpetra {
 
       RCP<const Teuchos::Comm<int> > comm = map.getComm ();
       const int numProcs = comm->getSize ();
-      const global_size_t nOverP = map.getGlobalNumElements () / numProcs;
       const LO LINVALID = Teuchos::OrdinalTraits<LO>::invalid();
       LookupStatus res = AllIDsPresent;
 
+      // Find the first initialized GID for search below
+      auto it = std::find_if(
+        allMinGIDs_.begin(),
+        allMinGIDs_.end()-1,
+        [](GO const gid) { return gid != std::numeric_limits<GO>::max(); }
+      );
+#ifdef HAVE_TPETRA_DEBUG
+      TEUCHOS_TEST_FOR_EXCEPTION(
+        it == allMinGIDs_.end()-1,
+        std::logic_error,
+        "Every min GID in the map is std::numeric_limits<GO>::max()!"
+      );
+#endif // HAVE_TPETRA_DEBUG
+      const GO i0 = std::distance(allMinGIDs_.begin(), it);
```

Comment:
You might want to make that an explicit cast, though it should be harmless as long as GO is signed.  Remember that Albany uses DTK 2.0 and thus uses `GO=unsigned long`.